### PR TITLE
Add locking based on issues found during race testing

### PR DESCRIPTION
**Pull request by shaleman**
Add locking to various data structures based on race detection in integration and unit tests

### CODE REVIEW

File: rpcHub/rpcHub.go [hunk 1]
```diff
@@ -113,6 +113,8 @@ func dialRpcClient(servAddr string, portNo uint16) (*rpc.Client, net.Conn) {
 // Get a client to the rpc server
 func Client(servAddr string, portNo uint16) *RpcClient {
 	clientKey := fmt.Sprintf("%s:%d", servAddr, portNo)
+	dbLock.Lock()
+	defer dbLock.Unlock()
```
Comment (jojimt) [inline rpcHub/rpcHub.go]: Should we use RLock for the lookup?

Comment (shaleman) [inline rpcHub/rpcHub.go]: It required a RLock() for reading it followed by a write lock to store new client.
I thought i'll simplify the code by taking a full lock at the top..

Comment (jojimt) [inline rpcHub/rpcHub.go]: Yes, figured that. If this is not called frequently, we are good. I was not sure of the calling pattern of Client().

Comment (jojimt) [inline rpcHub/rpcHub.go]: LGTM

**Pull request merged**